File: tests/unit/StrategyYVault/test_set_withdraw_fee.py
import brownie

ZERO_ADDRESS = "0x0000000000000000000000000000000000000000"


def test_set_withdraw_fee(accounts, strategyYVault):
    fee = 123
    tx = strategyYVault.setWithdrawFee(fee, { 'from': accounts[0] })

    assert strategyYVault.withdrawFee() == fee
    assert len(tx.events) == 1
    assert tx.events["SetWithdrawFee"].values() == [fee]


def test_set_controller_not_admin(accounts, strategyYVault):
    fee = 123
    with brownie.reverts("dev: !admin"):
        strategyYVault.setWithdrawFee(fee, {'from': accounts[1]})

File: test/unit/Vault/test-set-withdraw-fee.ts
import chai from "chai"
import BN from "bn.js"
import {VaultInstance} from "../../../types/Vault"
import {eq} from "../../util"
import _setup from "./setup"

contract("Vault", (accounts) => {
  const MIN_WAIT_TIME = 0

  const refs = _setup(accounts, MIN_WAIT_TIME)
  const {admin} = refs

  let vault: VaultInstance
  beforeEach(() => {
    vault = refs.vault
  })

  describe("setWithdrawFee", () => {
    it("should set withdraw fee", async () => {
      const fee = new BN(123)
      await vault.setWithdrawFee(fee, {from: admin})

      assert(eq(await vault.withdrawFee(), fee), "fee")
    })

    it("should reject if caller not admin", async () => {
      await chai.expect(vault.setWithdrawFee(123, {from: accounts[1]})).to.be.rejectedWith(
        "!admin"
      )
    })

    it("should reject min > max", async () => {
      await chai.expect(vault.setWithdrawFee(501, {from: admin})).to.be.rejectedWith(
        "withdraw fee > cap"
      )
    })
  })
})
